crates/steel-repl/src/highlight.rs

```diff
@@ -129,7 +129,9 @@ impl Validator for RustylineHelper {
                 Err(err) => {
                     unfinished = std::matches!(
                         err.ty,
-                        TokenError::IncompleteString | TokenError::IncompleteIdentifier
+                        TokenError::IncompleteString
+                            | TokenError::IncompleteIdentifier
+                            | TokenError::IncompleteComment
```

Comment:
:ringed_planet: Small unrelated fix: incomplete comments also count as multi-line input in the REPL.

Note that we cannot do that for unclosed escaped literals, b/c newlines are not accepted.